### PR TITLE
Adding cloudprovider check + meaningful message for invalid auth credentials

### LINKED ISSUE #3: Verify that AWS account is present
There needs to be a check made that verifies that the plugin has all appropriate parts including an AWS profile. 

The below error results from having empty AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY environment variables. A more meaningful message would be helpful. 

```
~#~#~#~DEPLOYING OPS REPO [cloudformation] SUCCESSFULLY COMPLETED~#~#~#~
2022-10-03 18:16:48,852 root         DEBUG 
Welcome to the CloudFormation BitOps plugin!
Deploying cloudformation... cloudformation - Found BitOps config
All dependent plugins found. Continuing with deployment..
cd cloudformation Root: /tmp/tmppqh6ucas/Dev/cloudformation
Running Cloudformation Template Validation
Running Cloudformation Deploy Stack
Checking if stack exists ...
STACK_EXISTS_OUTPUT Unable to locate credentials. You can configure credentials by running "aws configure".

Stack does not exist, creating ...
create-stack
Parameters file doesn't exist...
Waiting on cloudformation stack BitOps-Demo create-stack completion...
Finished cloudfromation action create-stack successfully !!!
```

### PR DESCRIPTION
**Pull request by PhillypHenning**
Adding a cloud provider check + meaningful usage message

fixes https://github.com/bitops-plugins/cloudformation/issues/3

### CODE REVIEW

**Comment by PhillypHenning** [comment]
Isolated if into `run_check.sh` for quick testing

```
[Bitovi] [cloudformation] $ ./run_check.sh 
AWS ACCESS KEY or SECRET isn't set. Check out our doc to understand what you're missing https://bitops.sh/examples/#docker-run-examples

[Bitovi] [cloudformation] $ echo $AWS_ACCESS_KEY_ID

[Bitovi] [cloudformation] $ echo $AWS_SECRET_ACCESS_KEY

[Bitovi] [cloudformation] $ prep_bitovi
Using Bitovi AWS credentials
[Bitovi] [cloudformation] $ echo $AWS_SECRET_ACCESS_KEY
secret_key_value
[Bitovi] [cloudformation] $ ./run_check.sh                  
[Bitovi] [cloudformation] $ export AWS_SECRET_ACCESS_KEY=
[Bitovi] [cloudformation] $ ./run_check.sh                    
AWS ACCESS KEY or SECRET isn't set. Check out our doc to understand what you're missing https://bitops.sh/examples/#docker-run-examples
```

**Comment by arm4b** [comment]
Provided thoughts on this in the original issue https://github.com/bitops-plugins/cloudformation/issues/3#issuecomment-1312232855

File: deploy.sh [hunk 1]
```diff
@@ -60,6 +60,13 @@ else
     fi
 fi
 
+aws sts get-caller-identity
+result=$?
+if [ $result != 0 ]; then
+    echo "AWS ACCESS KEY or SECRET isn't set. Check out our doc to understand what you're missing https://bitops.sh/examples/#docker-run-examples"
```
Comment (arm4b) [inline deploy.sh]: A more detailed error message and mentioning the ENV variables explicitly would be more helpful for someone hitting this error: 
```suggestion
    echo "AWS_ACCESS_KEY_ID or AWS_SECRET_ACCESS_KEY required for AWS authentication are not set or invalid."
    echo "Check out BitOps documentation to understand what you're missing https://bitops.sh/examples/#docker-run-examples"
```

File: deploy.sh [hunk 2]
```diff
@@ -60,6 +60,13 @@ else
...
+    exit $result
```
Comment (arm4b) [inline deploy.sh:68]: 👍

**Review by arm4b — CHANGES_REQUESTED** [review]

File: deploy.sh (hunk 2 above)
Comment (PhillypHenning) [reply deploy.sh:68]: Sorry does something need to be resolved here?

Comment (arm4b) [reply deploy.sh:68]: No, it's good. Just providing feedback on the part I liked.

**Review by arm4b — APPROVED** [review]

**Pull request merged**